changes/20248-non-windows-script-timeout

```diff
@@ -0,0 +1 @@
+- fixed issue where macOS scripts failed to timeout on long running commands
```

Comment:
Fix applies to Linux hosts as well? (So we should test/QA both.)

Reply:
💯 I'll add linux to the test plan